src/main/java/org/jboss/threads/Version.java

```diff
@@ -48,9 +51,17 @@ private Version() {
         }
         JAR_NAME = jarName;
         VERSION_STRING = versionString;
+
+        boolean logVersion = true;
         try {
-            Messages.msg.version(versionString);
+            logVersion = !"false".equalsIgnoreCase(System.getProperty(JBOSS_LOG_VERSIONS));
         } catch (Throwable ignored) {}
```

Comment:
Decided to be extra defensive to keep the original behavior even if `System.getProperty` throws something.

Reply:
+1 was just thinking what would happen if a security manager prevented the call.